config/routes.php

```diff
@@ -6,5 +6,45 @@
  * @var array $params
  */
 
+use Psr\Http\Message\ResponseFactoryInterface;
+use Yiisoft\DataResponse\Middleware\FormatDataResponseAsJson;
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+use Yiisoft\Validator\ValidatorInterface;
+use Yiisoft\Yii\Debug\Api\Middleware\ResponseDataWrapper;
+use Yiisoft\Yii\Gii\Controller\DefaultController;
+use Yiisoft\Yii\Middleware\IpFilter;
+
+if (!(bool)($params['yiisoft/yii-gii']['enabled'] ?? false)) {
+    return [];
+}
+
 return [
+    Group::create('/gii/api')
+         ->middleware(
+             static function (ResponseFactoryInterface $responseFactory, ValidatorInterface $validator) use ($params) {
+                 return new IpFilter(
+                     validator: $validator,
+                     responseFactory: $responseFactory,
+                     ipRanges: $params['yiisoft/yii-gii']['allowedIPs']
+                 );
+             }
+         )
+         ->middleware(FormatDataResponseAsJson::class)
+         ->middleware(ResponseDataWrapper::class)
+         ->namePrefix('gii/api/')
+         ->routes(
+             Route::get('/{generator}')
+                  ->action([DefaultController::class, 'get'])
+                  ->name('generator'),
+             Route::post('/{generator}/preview')
```

Comment:
Why `POST` is used? Seems like nothing is modified, so `GET` would be more appropriate?